tagstudio/src/qt/widgets/tag_box.py

```diff
@@ -120,6 +126,24 @@ def edit_tag(self, tag_id:int):
 		self.edit_modal.saved.connect(lambda: self.lib.update_tag(btp.build_tag()))
 		# panel.tag_updated.connect(lambda tag: self.lib.update_tag(tag))
 		self.edit_modal.show()
+	
+	
+	def delete_tag(self, tag_id:int):
+		def show_delete_prompt() -> bool:
+			result = QMessageBox.question(self, "Delete Tag", f"Are you sure you want to delete Tag {tag.name}?",
+								 QMessageBox.Yes | QMessageBox.No)
+			
+			return result == QMessageBox.Yes
+		
+		tag = self.lib.get_tag(tag_id)
+		shift_held = Qt.KeyboardModifier.ShiftModifier in QApplication.keyboardModifiers()
+		
+		if shift_held or show_delete_prompt():
+			self.lib.remove_tag(tag_id)
+			self.update_tags()
+			
+			if not shift_held:
+				QMessageBox.information(self, "Delete Tag", "Tag deleted.")
```

Comment:
Having to replicate this here and in tag_database makes me think there should be a more reusable way to implement these dialogs but I think that will be a whole codebase sweep that needs to happen so I don't think that's an issue for this PR.

Reply:
I considered putting this function in a place where I could reuse it without repeating myself. the other event listeners don't do this either though so I chose to be consistent with them